src/split.cc

```diff
@@ -0,0 +1,166 @@
+#include "split.h"
+
+#include <array>
```

Comment:
I feel like most (all?) of the uses of `std::array<size_t, Dims>` could instead be `range<Dims>`.

Reply:
Agreed for small/large chunk sizes; for chunk counts I feel like the `array` is a better semantic fit!